src/enc_effects.c

```diff
@@ -0,0 +1,366 @@
+#define SEQ_PL_BA_GIRA      1201        // Battle Giratina
+#define SEQ_PL_BA_BRAIN     1202        // Battle Frontier Brain
+#define SEQ_PL_BA_REGI      1204        // Battle Regi trio (not Regigigas)
+
+// need to research why these start at 0x000C
+#define OREBURGH_GYM_ENC        0x000C      // Roark
+#define ETERNA_GYM_ENC          0x000D      // Gardenia
+#define PASTORIA_GYM_ENC        0x000E      // Crasher Wake
+#define VEILSTONE_GYM_ENC       0x000F      // Maylene
+#define HEARTHOME_GYM_ENC       0x0010      // Fantina
+#define SNOWPOINT_GYM_ENC       0x0011      // Candice
+#define CANALAVE_GYM_ENC        0x0012      // Byron
+#define SUNYSHORE_GYM_ENC       0x0013      // Volkner
+#define ELITE_FOUR_ENC_1        0x0014      // Aaron
+#define ELITE_FOUR_ENC_2        0x0015      // Bertha
+#define ELITE_FOUR_ENC_3        0x0016      // Flint
+#define ELITE_FOUR_ENC_4        0x0017      // Lucian
+#define CHAMPION_ENC            0x0018      // Cynthia
+#define LEGENDARY_POKE_ENC_1    0x0019      // Roamers, Heatran, Regigigas, Darkrai, Shaymin
+#define LEGENDARY_POKE_ENC_2    0x001A      // Dialga, Palkia, Lake Guardians, Arceus
+#define GALACTIC_GRUNT_ENC      0x001B      // Grunts
+#define GALACTIC_LEADER_ENC     0x001C      // Mars, Jupiter, Saturn, Cyrus
+#define FRONTIER_ENC            0x001D      // Frontier / Wi-Fi Battles
+#define DOUBLE_ENC              0x001E      // 2v2s, wild battles with partner
+#define CATCHALL_ENC            0xFFFF      // catch-all?
+
+#define PAL_PARK_ZONE_ID        251         // pal park; important for certain wild legends
+
+typedef struct {
+    u16 cutinVFX;
+    u16 bgmSEQID;
+} EncEffectsDataPair;
+
+enum {
+    // Gym Leaders
+    ENCFX_OREBURGH_GYM_LEADER,
+    ENCFX_ETERNA_GYM_LEADER,
+    ENCFX_PASTORIA_GYM_LEADER,
+    ENCFX_VEILSTONE_GYM_LEADER,
+    ENCFX_HEARTHOME_GYM_LEADER,
+    ENCFX_SNOWPOINT_GYM_LEADER,
+    ENCFX_CANALAVE_GYM_LEADER,
+    ENCFX_SUNYSHORE_GYM_LEADER,
+
+    // Elite Four
+    ENCFX_ELITE_FOUR_1,
+    ENCFX_ELITE_FOUR_2,
+    ENCFX_ELITE_FOUR_3,
+    ENCFX_ELITE_FOUR_4,
+
+    // Champion
+    ENCFX_CHAMPION,
+
+    // Rival
+    ENCFX_RIVAL,
+
+    // Legendary Pokemon
+    ENCFX_SHAYMIN,
+    ENCFX_DIALGA_PALKIA,
+    ENCFX_UXIE_AZELF,
+    ENCFX_MESPRIT,
+    ENCFX_ARCEUS,
+    ENCFX_REGIGIGAS_HEATRAN,
+    ENCFX_CRESSELIA,
+    ENCFX_ARTICUNO_ZAPDOS_MOLTRES,
+    ENCFX_GIRATINA,
+    ENCFX_REGIROCK_REGICE_REGISTEEL,
+
+    // Team Galactic
+    ENCFX_GALACTIC_GRUNT,
+    ENCFX_GALACTIC_COMMANDER,
+    ENCFX_GALACTIC_LEADER,
+
+    // Other Special Trainers
+    ENCFX_FRONTIER_TRAINER,
+    ENCFX_LINK_WIFI_BATTLE,
+    ENCFX_DOUBLE_BATTLE_TRAINER,
+    ENCFX_DOUBLE_BATTLE_POKEMON,
+    ENCFX_FRONTIER_BRAIN,
+    ENCFX_DOUBLE_BATTLE_GYM_LEADER,
+
+    // Generic Trainers and Pokemon
+    ENCFX_GENERIC_TRAINER,
+    ENCFX_GENERIC_POKEMON,
+
+    ENCFX_NUM_ENTRIES
+};
+
+static const EncEffectsDataPair ENCOUNTER_EFFECTS[ENCFX_NUM_ENTRIES] = {
```

Comment:
All caps is generally reserved for enum values and defines, things that can be treated as regular variables should be capitalized like regular variables.